CMakeLists.txt

```diff
@@ -473,17 +475,28 @@ elseif(${PLAYER_TARGET_PLATFORM} STREQUAL "switch")
 		src/platform/switch/switch_clock.h
 		src/platform/switch/switch_ui.cpp
 		src/platform/switch/switch_ui.h)
-	target_compile_definitions(${PROJECT_NAME} PUBLIC __SWITCH__)
 else()
 	message(FATAL_ERROR "Invalid target platform")
 endif()
 
+if(3DS)
+	target_compile_definitions(${PROJECT_NAME} PUBLIC _3DS)
+elseif(VITA)
+	target_compile_definitions(${PROJECT_NAME} PUBLIC PSP2)
+elseif(NSWITCH)
+	target_compile_definitions(${PROJECT_NAME} PUBLIC __SWITCH__)
+endif()
```

Comment:
I dislike here that libretro dictates this. At least for 3DS we will need `if(3DS OR N3DS)` to stay compatible with devkitpro toolchain file or do something custom.

Reply:
I must improve the commit message. I don't plan 3DS support for libretro because our optimized port already runs badly.

This is actually not dictated by libretro. They hate CMake and only use there ugly makefiles. I just added this for consistency.

Must have looked in the wrong toolchain file. Will fix this, thanks

Reply:
fixed